### PR TITLE
WIP Updating the iOS/macOS flavors doc

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixes https://github.com/flutter/website/issues/10713,
for setting up flavors in iOS and macOS.

cc @qwadrox @jmagman

### CODE REVIEW

File: src/content/deployment/flavors.md [hunk 1]
```diff
@@ -114,6 +116,34 @@ build configurations already created.
 </li>
 </ol>
 
+### Manually update podfile
```
Comment (jmagman) [inline src/content/deployment/flavors.md:119]: Casing nit: Podfile
```suggestion
### Manually update Podfile
```

File: src/content/deployment/flavors.md [hunk 2]
```diff
@@ -114,6 +116,34 @@ build configurations already created.
...
+
+After running `flutter clean`, `flutter pub`, and `pod install`,
+you receive a build error indicating that it is unable to load
+the contents of a file list,
+you will need to manually update your podfile.
```
Comment (jmagman) [inline src/content/deployment/flavors.md:124]:
```suggestion
you will need to manually update your Podfile.
```

File: src/content/deployment/flavors.md [hunk 3]
```diff
@@ -114,6 +116,34 @@ build configurations already created.
...
+  'Release-free' => :release,
+  'Debug-prod' => :debug,
+  'Profile-prod' => :release,
+  'Release-prod' => :release,
+}
+```
+
+2. Run `flutter clean` and `flutter pub get`.
+
+3. Run `pod install`.
```
Comment (jmagman) [inline src/content/deployment/flavors.md:143]: Flutter developers should never need to run `pod install` (except for add-to-app).  If we have instructions to explicitly run it anywhere else in the docs, we should remove it.
You can delete steps 2 and 3 and go right to "Build the app."

**Review by jmagman — CHANGES_REQUESTED** [review]
https://github.com/flutter/website/issues/10713 is reporting an issue in adding a flavor (prod) that isn't described in these docs.  These instructions are only for adding the "free" flavor, they don't say anything about prod.  

And we already have the correct instructions for updating the Podfile here:
https://docs.flutter.dev/deployment/flavors#plugin-configurations

**Comment by jmagman** [comment]
Reading it again, the fix for https://github.com/flutter/website/issues/10713 would be to just add the instructions for updating the xcconfig, as they say in their workaround section:
```diff
- #include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"
+ #include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug-free.xcconfig"
```
Example from our integration docs:
https://github.com/flutter/flutter/blob/32da263b1d5590a98e03d0ea1374f85e0c18de3f/dev/integration_tests/flavors/ios/Flutter/Debug.xcconfig#L3

**Comment by qwadrox** [comment]
> Reading it again, the fix for #10713 would be to just add the instructions for updating the xcconfig, as they say in their workaround section:
> 
> ```diff
> - #include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"
> + #include? "Pods/Target Support Files/Pods-Runner/Pods-Runner.debug-free.xcconfig"
> ```
> 
> Example from our integration docs: https://github.com/flutter/flutter/blob/32da263b1d5590a98e03d0ea1374f85e0c18de3f/dev/integration_tests/flavors/ios/Flutter/Debug.xcconfig#L3

What is the purpose of xcconfig files?
If I have 3 scheme like prod, uat, dev, should I duplicate the 'Debug' => :debug block in podfile 3 times? like  'Debug-prod' => :debug, ... etc. 
First I thought I have to have 3 line per scheme as well in the xcconfig files: #include? like 
```
"Pods/Target Support Files/Pods-Runner/Pods-Runner.debug-prod.xcconfig"
"Pods/Target Support Files/Pods-Runner/Pods-Runner.debug-dev.xcconfig"
"Pods/Target Support Files/Pods-Runner/Pods-Runner.debug-uat.xcconfig"
```
But it seems one is enough

**Comment by jmagman** [comment]
> What is the purpose of xcconfig files?

It's an Xcode concept to add build configurations to your project:
https://developer.apple.com/documentation/xcode/adding-a-build-configuration-file-to-your-project

You likely only needed one because they are all identical.

**Comment by qwadrox** [comment]
> > What is the purpose of xcconfig files?
> 
> It's an Xcode concept to add build configurations to your project:
> https://developer.apple.com/documentation/xcode/adding-a-build-configuration-file-to-your-project
> 
> You likely only needed one because they are all identical.

I understand now. Thanks. 
The error occurred because I renamed both schemes. I renamed the original Runner scheme to 'prod', which caused the imported Pod generated config to be incorrectly referenced in Debug.xconfig.

The original documentation is correct; it only advises creating a new scheme, leaving the Runner scheme unchanged. This maintains the correct reference paths in Debug.xconfig.

It would be only beneficial to add a cautionary note to the docs: 

"If you rename the original Runner scheme, ensure you update Debug.xcconfig and Release.xcconfig in the ios/Flutter directory to reference the newly generated Pods/Target Support Files/Pods-Runner xcconfig file."

**Comment by atsansone** [comment]
@sfshaza2 : I have a larger refactor of this whole page in flight. I'll integrate your changes to it and then we can close this PR. Does that work for you?

**Comment by sfshaza2** [comment]
@atsansone, I'd have preferred if you had talked about this with me ahead of time.

**Pull request closed without merging**